R/gather.R

```diff
@@ -53,6 +53,22 @@
 #' @param factor_key If `FALSE`, the default, the key values will be
 #'   stored as a character vector. If `TRUE`, will be stored as a factor,
 #'   which preserves the original ordering of the columns.
+#'
+#' @details
```

Comment:
I like the intent here, but I think:

* It should be it's own section, e.g. `@section Compared to `pivot_long()`
* It can be much shorter, and you don't need to reference the tidy data paper (which uses reshape2 anyway)
* Needs to link to `pivot_long()` with `[pivot_long()]`
* It should be a bit more assertive about switching to `pivot_long()` - `gather()` isn't going away but `pivot_long()` is more featureful and should be easier for most people to learn.

Reply:
I agree with all these comments, and have pushed these changes